cmd/rpcdaemon/commands/erigon_receipts.go

```diff
@@ -40,6 +50,165 @@ func (api *ErigonImpl) GetLogsByHash(ctx context.Context, hash common.Hash) ([][
 	return logs, nil
 }
 
+// GetLogs implements eth_getLogs. Returns an array of logs matching a given filter object.
+func (api *ErigonImpl) GetLogs(ctx context.Context, crit filters.FilterCriteria) ([]*types.Log, error) {
+	var begin, end uint64
+	logs := []*types.Log{}
+
+	tx, beginErr := api.db.BeginRo(ctx)
+	if beginErr != nil {
+		return logs, beginErr
+	}
+	defer tx.Rollback()
+
+	if crit.BlockHash != nil {
+		number := rawdb.ReadHeaderNumber(tx, *crit.BlockHash)
+		if number == nil {
+			return nil, fmt.Errorf("block not found: %x", *crit.BlockHash)
+		}
+		begin = *number
+		end = *number
+	} else {
+		// Convert the RPC block numbers into internal representations
+		latest, err := rpchelper.GetLatestBlockNumber(tx)
+		if err != nil {
+			return nil, err
+		}
+
+		begin = latest
+		if crit.FromBlock != nil {
+			if crit.FromBlock.Sign() >= 0 {
+				begin = crit.FromBlock.Uint64()
+			} else if !crit.FromBlock.IsInt64() || crit.FromBlock.Int64() != int64(rpc.LatestBlockNumber) {
+				return nil, fmt.Errorf("negative value for FromBlock: %v", crit.FromBlock)
+			}
+		}
+		end = latest
+		if crit.ToBlock != nil {
+			if crit.ToBlock.Sign() >= 0 {
+				end = crit.ToBlock.Uint64()
+			} else if !crit.ToBlock.IsInt64() || crit.ToBlock.Int64() != int64(rpc.LatestBlockNumber) {
+				return nil, fmt.Errorf("negative value for ToBlock: %v", crit.ToBlock)
+			}
+		}
+	}
+	if end < begin {
+		return nil, fmt.Errorf("end (%d) < begin (%d)", end, begin)
+	}
+	if end > roaring.MaxUint32 {
```

Comment:
How it can happen? Is there som uint underflow in logic above?

Reply:
The method in roaring `AddRange` The function uses 64-bit parameters even though a Bitmap stores 32-bit values because it is allowed and meaningful to use [0,uint64(0x100000000)) as a range. 
`
if end > roaring.MaxUint32 {
		return nil, fmt.Errorf("end (%d) > MaxUint32", end)
	}
	blockNumbers := roaring.New()
	blockNumbers.AddRange(begin, end+1) // [min,max)
        ...
}

func (rb *Bitmap) AddRange(rangeStart, rangeEnd uint64) {
	if rangeStart >= rangeEnd {
		return
	}
	if rangeEnd-1 > MaxUint32 {
		panic("rangeEnd-1 > MaxUint32")
	}
        ...
}

`
The code should check or use a 32-bit integer parameter.

Reply:
https://github.com/RoaringBitmap/roaring/blob/master/roaring.go#L1484

Reply:
The code in roaring has a limit. But variables `end` and `start` are type uint64. Maybe use uint32 at the beginning?

Reply:
I think yes. Or convert bitmap to bitmap64 (not easy - need to change db format), or convert rangeEnd to uint32 (if such a big value is not an underflow).

Reply:
because such a big value - maybe not an error, maybe it just way to say: "give me all blocks > N"

Reply:
Ok. I changed it to the latest block when rangeEnd is too big. Plz check it.